gamla/functional_generic.py

```diff
@@ -133,7 +133,20 @@ def lazyjuxt(*args, **kwargs):
     return lazyjuxt
 
 
-juxt = compose(after(tuple), lazyjuxt)
+def juxt(*funcs):
+    if _any_is_async(funcs):
+        funcs = tuple(map(after(to_awaitable), funcs))
+
+        async def inner_juxt_async(*args, **kwargs):
+            return await asyncio.gather(*map(lambda f: f(*args, **kwargs), funcs))
+
+        return compose(after(tuple), inner_juxt_async)
```

Comment:
why do we need to compose with tuple again?

Reply:
I think `asyncio.gather` already returns a `tuple`

Reply:
it doesn't, ir returns a list. i can either change the test to assert a list, or make this return a tuple